source/game_sa/CFire.cpp

```diff
@@ -1,6 +1,267 @@
 #include "StdInc.h"
 
+void CFire::InjectHooks()
+{
+    //ReversibleHooks::Install("CFire", "Constructor", 0x539D90, &CFire::Constructor);
+    //ReversibleHooks::Install("CFire", "Destructor", 0x538BA0, &CFire::Destructor);
+    ReversibleHooks::Install("CFire", "Initialise", 0x538B30, &CFire::Initialise);
+    ReversibleHooks::Install("CFire", "CreateFxSysForStrength", 0x539360, &CFire::CreateFxSysForStrength);
+    ReversibleHooks::Install("CFire", "Extinguish", 0x5393F0, &CFire::Extinguish);
+    ReversibleHooks::Install("CFire", "ProcessFire", 0x53A570, &CFire::ProcessFire);
+}
+
+CFire::CFire()
+{
+}
+
+CFire::~CFire()
+{
+}
+
+CFire* CFire::Destructor()
+{
+    this->CFire::~CFire();
+    return this;
+}
+
+CFire* CFire::Constructor()
+{
+    this->CFire::CFire();
+    return this;
+}
+
+void CFire::Initialise()
+{
+    // Originally m_nFlags = (m_nFlags & 0xF4) | 0x14; - Clear 1st, 2nd, 4th and set 3rd, 5th bits (1-based numbering)
+    m_nFlags = {
+        false, // bActive
+        false, // bCreatedByScript
+        true,  // bMakesNoise
+        false, // bBeingExtinguished
+        true   // bFirstGeneration
+    };
```

Comment:
```suggestion
    bActive = false;
    bCreatedByScript = false;
    bMakesNoise = true;
    bBeingExtinguished = false;
    bFirstGeneration = true;
```

We shouldn't use m_nFlags, instead use bit fields like R*. In CFire.h, you can change take the bit fields out of the struct and remove m_nFlags:
```
uint8_t bActive : 1;
uint8_t bCreatedByScript : 1;
uint8_t bMakesNoise : 1;
uint8_t bBeingExtinguished : 1;
uint8_t bFirstGeneration : 1;
```

Reply:
Will do.